react/CheckPermission.tsx

```diff
@@ -9,28 +9,32 @@ interface Props {
   roles: string[]
   AllowedContent?: ElementType
   DisallowedContent?: ElementType
+  LoadingContent?: ElementType
 }
 
 function CheckPermission({
   roles = [],
   AllowedContent,
   DisallowedContent,
+  LoadingContent,
 }: Props) {
   const { data, called, error, loading } = useQuery(getPermissions, {
     ssr: false,
     skip: !roles.length,
   })
 
-  if (!roles.length || !data) {
+  if (error) {
+    console.error('CheckPermission error:', error)
+
     return null
   }
 
-  if ((called && loading) || error) {
-    if (error !== undefined) {
-      console.error('CheckPermission error:', error)
-    }
-
-    return null
+  if (!roles.length || !data || (called && loading)) {
```

Comment:
Just a small thing I've noticed now: before this change we'd only treat the content as loading when `(!roles.length || !data) && (called && loading)` (nested ifs). But here your check uses `||` instead of `&&`, which might cause different results in certain edge cases. For example, when `!roles.length` is true we'll now show the loading indicator even though `loading` will be false. In that case we'd show something else on the screen before this change.

So, to be safe, can you replace the operator, like this:

```suggestion
  if (!roles.length || !data && (called && loading)) {
```

Reply:
You are absolutely right @mairatma. I just pushed a more secure implementation for edge cases, which worked perfectly in development workspace..